services:
  django:
    platform: linux/amd64
    build:
      context: .
      dockerfile: ./compose/local/django/Dockerfile
    depends_on:
      - postgres
    volumes:
      - .:/app:z
      - ../ds-caselaw-custom-api-client:/apiclient
    init: true
    environment:
      DATABASE_URL: "postgres://postgres:admin@postgres:5432/ds_judgments_editor_ui"
      DJANGO_SETTINGS_MODULE: "config.settings.local"
      SECRET_KEY: local_dev_secret_key
      SECURE_SSL_REDIRECT: "false"
    env_file:
      - ./.env
    ports:
      - "3000:3000"
    # do nothing forever - exec commands elsewhere
    command: tail -f /dev/null

  postgres:
    image: postgres:15.14@sha256:d83779f5de2a1f1c54af5e806c0092f8de03cf2801b340a2145b93b02e72bd65
    ports:
      - "5432:5432"
    environment:
      POSTGRES_DB: ds_judgments_editor_ui
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: admin
      POSTGRES_HOST: postgres
      POSTGRES_PORT: 5432
      PGDATABASE: ds_judgments_editor_ui
      PGUSER: postgres
      PGPASSWORD: admin
      POSTGRES_HOST_AUTH_METHOD: trust
    volumes:
      - ./database_backups:/backups:z

  e2e_tests:
    container_name: e2e_tests_container
    build:
      context: ./e2e_tests
      dockerfile: "../compose/local/e2e_tests/Dockerfile"

    profiles: [e2e_tests]
    volumes:
      - "./e2e_tests:/app:z"
    tty: true
